src/core/reference/include/openvino/reference/concat.hpp

```diff
@@ -15,14 +16,8 @@ void concat(const std::vector<const char*>& args,
             const std::vector<Shape>& in_shapes,
             const Shape& out_shape,
             int64_t concatenation_axis,
-            size_t elem_size);
-
-void concat(const std::vector<const std::string*>& args,
-            std::string* out,
-            const std::vector<Shape>& in_shapes,
-            const Shape& out_shape,
-            int64_t concatenation_axis,
-            size_t);
+            size_t elem_size,
+            const ov::element::Type& elem_type = ov::element::Type_t::undefined);
```

Comment:
```suggestion
            const ov::element::Type& elem_type = ov::element::undefined);
```

```suggestion
            const ov::element::Type& elem_type = ov::element::dynamic);
```
Just to consider, as indicates data is any but defined type.

Reply:
If `element_type` is not provided to the function, the `elem_size` is the only indicator of the type, the `undefined` as default seems more reasonable to me, as the definition of `dynamic` doesn't fit here either.